### PR TITLE
✨ add hardcoded genesis block header

### LINKED ISSUE #29: Hardcode genesis block
The genesis block must be hardcoded and registered as the first block of the chain.

Any block with the hash of the genesis block must bypass validation rules.

See https://github.com/bitcoin-stark/khepri-starknet/blob/main/docs/bitcoin/block_header_validation_rules.md

### PR DESCRIPTION
**Pull request by ofux**
Closes #29

### CODE REVIEW

**Review by AbdelStark — APPROVED** [review]
lgtm

File: src/header/library.cairo [hunk 1]
```diff
@@ -193,13 +200,50 @@ namespace BlockHeaderVerifier:
...
+            bits=0x1d00ffff,
+            prev_block=Uint256(0, 0),
+            merkle_root=genesis_merkel_root,
+        )
+        return (genesis_block_header)
+    end
+
+    func felt_to_uint256{range_check_ptr}(x) -> (res : Uint256):
+        let (hi, lo) = split_felt(x)
+        return (Uint256(lo, hi))
```
Comment (tekkac) [inline src/header/library.cairo:247]: This function should  be imported from another library after refactor, right?

File: src/header/library.cairo [hunk 2]
```diff
@@ -193,13 +200,50 @@ namespace BlockHeaderVerifier:
...
+    end
+
+    # Returns the hardcoded genesis block.
+    # See https://github.com/bitcoin/bitcoin/blob/master/src/chainparams.cpp
+    # and https://www.blockchain.com/btc/block/000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f
+    func genesis_block_header{range_check_ptr}(genesis_block_header : BlockHeader):
+        let (genesis_hash : Uint256) = felt_to_uint256(
+            0x19d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f
+        )
+        let (genesis_merkel_root : Uint256) = felt_to_uint256(
```
Comment (tekkac) [inline src/header/library.cairo]:
```suggestion
        let (genesis_merkle_root : Uint256) = felt_to_uint256(
```

File: src/header/library.cairo [hunk 3]
```diff
@@ -193,13 +200,50 @@ namespace BlockHeaderVerifier:
...
+            0x4a5e1e4baab89f3a32518a88c31bc87f618f76673e2cc77ab2127b7afdeda33b
+        )
+        let genesis_block_header = BlockHeader(
+            hash=genesis_hash,
+            version=1,
+            timestamp=1231006505,
+            nonce=2083236893,
+            bits=0x1d00ffff,
+            prev_block=Uint256(0, 0),
+            merkle_root=genesis_merkel_root,
```
Comment (tekkac) [inline src/header/library.cairo]:
```suggestion
            merkle_root=genesis_merkle_root,
```

**Review by tekkac — APPROVED** [review]
lgtm

File: src/header/library.cairo (hunk 1 above)
Comment (ofux) [reply src/header/library.cairo:247]: yes indeed, the refactor will come with my PR about target rule

**Pull request merged**